### ISSUE TITLE
How to delete (not cut)?

### ISSUE THREAD

**Issue by ericsaboia** [t9md/atom-vim-mode-plus#473]
Hello,

I'm not very familiar with vim, but from [stackoverflow](http://stackoverflow.com/questions/11993851/how-to-delete-not-cut-in-vim) I saw that would be possible to delete (d) without copy (yank) by adding the mapping:

```
nnoremap <leader>d "_d
vnoremap <leader>d "_d
vnoremap <leader>p "_dP
```

Is it something possible on atom-vim-mode? 

Thanks for the help :)

**Comment by bronson** [comment]
There's no good way to [compose commands](http://flight-manual.atom.io/behind-atom/sections/keymaps-in-depth/#composed-commands) or replay macros in Atom.

You could try writing a `delete-without-yank` command in your init.coffee file.  This is an incomplete guess:

```cs
atom.commands.add 'atom-text-editor', 'custom:delete-without-yank', ->
  view = atom.views.getView(atom.workspace.getActiveTextEditor())
  # TODO: I could not figure out how to set the register to '-'
  # atom.commands.dispatch(view, "vim-mode-plus:set-register-name")
  atom.commands.dispatch(view, "vim-mode-plus:delete")
```

Once that command works, you'd bind it to whatever keypress you'd like.  Assuming your mapleader is `\`, your keybinding would look something like:

```cson
 'atom-text-editor.vim-mode-plus:not(.insert-mode)':
  '\ d': 'custom:delete-without-yank'
```

If setting the register is too difficult, maybe you could override the `Delete` class the way MoveUp and MoveDown were overridden here: https://github.com/t9md/atom-vim-mode-plus/wiki/ExtendVimModePlusInInitFile#simple-moveupmovedown-motion

**Comment by t9md** [comment]
@ericsaboia 

## Short answer

What you saw in stackoverflow is using black whole register(`_`).
You can specify what register to use for `d`, and by specifying `_` register for `d` you save nothing to register, this is how to achieve delete(not cut) in vim.

So when you want to delete line, keystroke would be `" _ d d`.
For delete word `" _ d w` or `" _ d i w` etc..

## How to shorten `" _` part.

Yes repeating `" _` every time is tedious, I understand that.
This is why @bronson trying to do..
I'll too try to find easy way.

**Comment by t9md** [comment]
With very experimental my [keystroke](https://atom.io/packages/keystroke) package ( I haven't maintained actively because of lack of time).

This package provides the way to define new command for specified keystroke.

in `init.coffee`
```coffeescript
  keystroke:
    commands: [
      {
        name: "set-blackhole-register"
        keystroke: '" _' #
      }
```

in `keymap.cson`
```coffeescript
'atom-text-editor.vim-mode-plus.normal-mode':
  "\\": "keystroke:set-blackhole-register"
```

Then `\ d d` will delete line but not update your register.
Since `\` keystroke use blackhole register(`_`).

**Comment by ericsaboia** [comment]
wow, thanks for the quick feedback guys! I'll give it a shot and post the results back here.

**Comment by t9md** [comment]
Here is the way to verion of @bronson tried.

https://github.com/t9md/atom-vim-mode-plus/wiki/ExtendVimModePlusInInitFile#advanced-deletewithbackholeregister

**Comment by ericsaboia** [comment]
It worked perfectly. Thank you!

**Comment by philgyford** [comment]
I've tried @t9md's code from the wiki (thanks!) but it doesn't change the behaviour for me. (Atom 1.17.2, vim-mode-plus 0.93.0, MacOS 10.12.4) I've tried restarting Atom.

I've put the `consumeService` function and the DeleteWithBackholeRegister use of it into `init.coffee` and added 

    'atom-text-editor.vim-mode-plus:not(.insert-mode)':
      '\\ d': 'vim-mode-plus-user:delete-with-backhole-register'

to my keymap.

But if I:

* type something
* hit `Escape`
* select all
* press `d`
* press `i`
* then `cmd-v`

it outputs the text that was deleted, rather than what was previously on the clipboard. The same if I do `p` rather than `i` and `cmd-v`. Have I missed something out in the setup?
